src/sources/kafka.rs

```diff
@@ -179,6 +183,22 @@ fn kafka_source(
                                 log.insert(offset_key, Value::from(msg.offset()));
                             }
 
+                            if let Some(headers_key) = &headers_key {
+                                let mut headers_map = BTreeMap::new();
+                                if let Some(headers) = msg.headers() {
+                                    for i in 0..headers.count() {
+                                        let header = headers.get(i).unwrap();
+                                        headers_map.insert(
+                                            header.0.to_string(),
+                                            Value::from(
+                                                String::from_utf8_lossy(header.1).to_string(),
```

Comment:
I might recommend just storing them into `Event` as `Bytes` `Value::from<String>` actually just converts it back to bytes.

https://github.com/timberio/vector/blob/ab0beeeb5b2e7736816a529dce8f18bef8342414/src/event/value.rs#L54-L58